Test redirects by specified HTTP status code

timothyjensen opened the pull request:
Simplifies testing additional HTTP status codes, such as `302` and `303`.

Closes #9

timothyjensen committed:
Test redirects by specified HTTP status code

Simplifies testing additional HTTP status codes, such as `302` and `303`.

Closes #9

GaryJones commented on src/Redirects.php:
```diff
@@ -86,6 +86,18 @@ public function seeTemporaryRedirectBetween($oldUrl, $newUrl)
+	 * @param string $newUrl     Relative or absolute URL of redirect destination.
+	 * @param int    $statusCode HTTP Status code to check against.
+	 */
+	public function seeOtherRedirectBetween($oldUrl, $newUrl, $statusCode)
```
When I suggested `seeOtherRedirectBetween()`, it was because the the 303 HTTP Status Code is called _Other Redirect_.

As such, line 98 would be hard-coded to 303, like the others are 307 and 301.

As it stands, this method is just a public wrapper to the private `seeRedirectBetween()`, so the simpler solution would be to change the access of `seeRedirectBetween()` so it's usable outside of the class.

If this happens, then the documentation should also be updated as well, including noting that the existing `see*RedirectBetween()` methods would now just be convenience methods.

timothyjensen replied:
Ah, my mistake. I think your proposed solution makes the most sense so I will go ahead with making `seeRedirectBetween()` public and updating the documentation accordingly.

timothyjensen committed:
Changed `seeRedirectBetween()` to public

- Removed `seeOtherRedirectBetween()`
- Updated docblocks for `see*RedirectBetween()` to show they are convenience methods
- Updated README with the addition of `seeRedirectBetween()` to the Public API.

See #9

GaryJones commented on README.md:
````diff
@@ -159,9 +159,27 @@ $I->followRedirects(false);
+Check that the specified HTTP Status is returned with the correct Location URL. Fails if either is missing, or `Location` header value does not match the `$url`. Automatically avoids following redirects.
+
+```php
+$I->seeRedirectBetween('company/financial-strength-and-security.cfm', 'company/financial-security', 302 );
````
Space after the `302` needs removing.

timothyjensen committed:
Removed extra space
